go/vt/vtadmin/api.go

```diff
@@ -246,22 +246,24 @@ func (api *API) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 				err = json.Unmarshal(decoded, &clusterJSON)
 				if err == nil {
 					clusterID := clusterJSON.ClusterName
-					c, err := cluster.Config{
-						ID:            clusterID,
-						Name:          clusterID,
-						DiscoveryImpl: "dynamic",
-						DiscoveryFlagsByImpl: cluster.FlagsByImpl{
-							"dynamic": map[string]string{
-								"discovery": string(decoded),
+					if _, exists := api.clusterMap[clusterID]; !exists {
+						c, err := cluster.Config{
+							ID:            clusterID,
+							Name:          clusterID,
+							DiscoveryImpl: "dynamic",
+							DiscoveryFlagsByImpl: cluster.FlagsByImpl{
+								"dynamic": map[string]string{
+									"discovery": string(decoded),
+								},
 							},
-						},
-					}.Cluster()
-					if err == nil {
-						api.clusterMap[clusterID] = c
-						api.clusters = append(api.clusters, c)
-						err = api.clusterCache.Add(clusterID, c, 24*time.Hour)
-						if err != nil {
-							log.Infof("could not add dynamic cluster %s to cluster cache: %+v", clusterID, err)
+						}.Cluster()
+						if err == nil {
+							api.clusterMap[clusterID] = c
+							api.clusters = append(api.clusters, c)
+							err = api.clusterCache.Add(clusterID, c, 24*time.Hour)
```

Comment:
i was looking at the docs, and i think we should use `cache.DefaultExpiration` which will cover us in case we update the one constant without the other

Reply:
also i filed https://github.com/vitessio/vitess/projects/12#card-80225244

Reply:
That's going to be cool!